umap/static/umap/js/umap.forms.js

```diff
@@ -804,18 +804,67 @@ L.FormBuilder.FacetSearch = L.FormBuilder.Element.extend({
       label = L.DomUtil.create('label', '', property_li)
     input.type = 'checkbox'
     input.id = `checkbox_${this.name}_${value}`
-    input.checked = this.get().includes(value)
+    input.checked = this.get()['choices'].includes(value)
     input.dataset.value = value
     label.htmlFor = `checkbox_${this.name}_${value}`
     label.innerHTML = value
     L.DomEvent.on(input, 'change', (e) => this.sync())
   },
 
   toJS: function () {
-    return [...this.ul.querySelectorAll('input:checked')].map((i) => i.dataset.value)
+    return {
+      'type': 'checkbox',
+      'choices': [...this.ul.querySelectorAll('input:checked')].map((i) => i.dataset.value)
+    }
   },
 })
 
+L.FormBuilder.FacetSearchDate = L.FormBuilder.Element.extend({
+  build: function () {
+    this.container = L.DomUtil.create('div', 'umap-facet', this.parentNode);
+    const min = this.options.criteria['min'];
+    const max = this.options.criteria['max'];
+
+    // Create labels for min and max inputs
+    this.minLabel = L.DomUtil.create('label', '', this.container);
+    this.minLabel.innerHTML = 'Start';
+    this.minLabel.htmlFor = `date_${this.name}_min`;
+
+    this.minInput = L.DomUtil.create('input', '', this.container);
+    this.minInput.type = 'datetime-local';
```

Comment:
Nice, I didn't know about [that input type](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/datetime-local) 👍